sky/skylet/providers/gcp/node.py

```diff
@@ -289,6 +289,42 @@ def create_instance(
         Returns a tuple of (result, node_name).
         """
         return
+    
+    def resize_disk(
+        self, base_config: dict, node_id: str, resources, wait_for_operation: bool = True
+    ) -> Tuple[dict, str]:
+        """Resize a Google Cloud disk based on the provided configuration."""
+        # Import required libraries
+        import google.cloud.compute_v1 as gcloud_compute
```

Comment:
Instead of `compute_v1`, can we use the `self.resource` (which is a GCP Compute resource) to align the other functions' implementation? Reference: https://github.com/skypilot-org/skypilot/blob/7a54940c0022b6d28919bd61b8443b9b72654ca7/sky/skylet/providers/gcp/node_provider.py#L86-L101

Reply:
In the latest commit, I've ensured that the resizing takes effect for both existing and new instances. Regarding `compute_v1`, I’m using it because I'm under the impression that GCP Compute resource might not offer APIs specifically for disk resizing and fetching the disk name. However, if there are functionalities I might have overlooked, please let me know, and I'd be happy to align the implementation with other functions.

Reply:
Just tried this code snippet and it works well for me:
```python
resource = discovery.build('compute', 'v1')

resp = resource.disks().resize(
    project=project_id,
    zone=zone,
    disk=disk_name,
    body={
        "sizeGb": "300",
    },
).execute()

print(resp)
```

Reply:
Also, to get disk name attached to the instance:

```python
resp = resource.instances().get(
    project=project_id,
    zone=zone,
    instance=instance_name
).execute()

print(resp["disks"])
```

Reply:
Hello @cblmemo, I've updated our code to use `googleapiclient` instead of `compute_v1`, and I've also adjusted it to resize disk exclusively for custom images, along with implementing a few small tweaks. However, I’m not sure if we should allow users to adjust the disk size after a restart.

Reply:
wdym by `adjust the disk size after a restart`? IIUC in current implementation, it will immediately got resized once the instance is launched?

Reply:
To provide further clarification, the issue arises after the execution of the following commands:
```
sky launch -c custom-image --cloud gcp --image-id projects/..../global/machineImages/... --disk-size 200
sky stop custom-image
sky start custom-image
```

Error:
```Got googleapiclient.errors.HttpError: <HttpError 400 when requesting https://compute.googleapis.com/compute/v1/projects/.../zones/us-central1-a/disks/ray-custom-image-d9a3-head-72d20b4d-compute/resize?alt=json returned "Invalid value for field 'sizeGb': '200'. New disk size '200' GiB must be larger than existing size '200' GiB.". Details: "[{'message': "Invalid value for field 'sizeGb': '200'. New disk size '200' GiB must be larger than existing size '200' GiB.", 'domain': 'global', 'reason': 'invalid'}]">```